kinto/core/storage/testing.py

```diff
@@ -525,6 +525,24 @@ def test_get_all_handle_all_pagination_rules(self):
         self.assertEqual(total_records, 10)
         self.assertEqual(len(records), 4)
 
+    def test_records_filtered_when_searched_for_startswith_on_string_field(self):
+        self.create_record({'name': 'barfoo'})
+        self.create_record({'name': 'foobar'})
+        self.create_record({'name': 'FOOBAR'})
+
+        filters = [Filter('name', 'FoO', utils.COMPARISON.STARTSWITH)]
+        results, count = self.storage.get_all(filters=filters, **self.storage_kw)
+        self.assertEqual(len(results), 2)
+
+    def test_records_filtered_when_searched_for_endswith_on_string_field(self):
+        self.create_record({'name': 'barfoo'})
+        self.create_record({'name': 'foobar'})
+        self.create_record({'name': 'FOOBAR'})
+
+        filters = [Filter('name', 'BaR', utils.COMPARISON.ENDSWITH)]
+        results, count = self.storage.get_all(filters=filters, **self.storage_kw)
+        self.assertEqual(len(results), 2)
+
```

Comment:
As @n1k0  pointed out, we should add tests when stored objects contain integers, lists etc. We've seen numerous bugs (in PG) when relying implicit casts...